prov/usnic/src/usdf_domain.c

```diff
@@ -365,43 +283,51 @@ usdf_domain_open(struct fid_fabric *fabric, struct fi_info *info,
 		if (udp->dom_dev != NULL) {
 			usd_close(udp->dom_dev);
 		}
-		usdf_dom_rdc_free_data(udp);
 		free(udp);
 	}
 	return ret;
 }
 
+/* In pre-1.4, the domain name was NULL.
+ *
+ * There used to be elaborate schemes to try to preserve this pre-1.4
+ * behavior.  In Nov 2019 discussions, however, it was determined that
+ * we could rationalize classifying this as buggy behavior.
+ * Specifically: we should just now always return a domain name --
+ * even if the requested version is <1.4.
+ *
+ * This greatly simplifies the logic here, and also greatly simplifies
+ * layering with the rxd provider.
+ */
 int usdf_domain_getname(uint32_t version, struct usd_device_attrs *dap,
 			char **name)
 {
 	int ret = FI_SUCCESS;
-	char *buf = NULL;
+	char *buf;
 
-	if (FI_VERSION_GE(version, FI_VERSION(1, 4))) {
-		buf = strdup(dap->uda_devname);
-		if (!buf) {
-			ret = -errno;
-			USDF_DBG("strdup failed while creating domain name\n");
-		}
+	buf = strdup(dap->uda_devname);
+	if (!buf) {
+		ret = -errno;
+		USDF_DBG("strdup failed while creating domain name\n");
 	}
 
 	*name = buf;
 	return ret;
 }
 
-/* In pre-1.4 the domain name was NULL. This is unfortunate as it makes it
- * difficult to tell whether providing a name was intended. In this case, it can
- * be broken into 4 cases:
+/* In pre-1.4, the domain name was NULL.
+ *
+ * There used to be elaborate schemes to try to preserve this pre-1.4
+ * behavior.  In Nov 2019 discussions, however, it was determined that
+ * we could rationalize classifying this as buggy behavior.
+ * Specifically: we should just now always return a domain name --
+ * even if the requested version is <1.4.
+ *
+ * This greatly simplifies the logic here, and also greatly simplifies
+ * layering with the rxd provider.
  *
- * 1. Version is greater than or equal to 1.4 and a non-NULL hint is provided.
- *    Just do a string compare.
- * 2. Version is greater than or equal to 1.4 and provided hint is NULL.  Treat
- *    this as _valid_ as it could be an application requesting a 1.4 domain name
- *    but not providing an explicit hint.
- * 3. Version is less than 1.4 and a name hint is provided.  This should always
- *    be _invalid_.
- * 4. Version is less than 1.4 and name hint is NULL. This will always be
- *    _valid_.
+ * Hence, if a hint was provided, check the domain name (that we now
+ * always have) against the hint.
```

Comment:
What does this routine do?  Why does no hint mean return true?

Reply:
It checks to see if the domain name supplied in a hint matches our current domain name.  Added a comment line about this.